python/pydantic_core/_pydantic_core.pyi

```diff
@@ -55,18 +55,43 @@ _T = TypeVar('_T', default=Any, covariant=True)
 
 @final
 class Some(Generic[_T]):
+    """
+    Similar to Rust's [`Option::Some`](https://doc.rust-lang.org/std/option/enum.Option.html) type, this
+    identifies a value as being present, and provides a way to access it.
+
+    Generally used in a union with `None` to different between "some value which could be None" and no value.
+    """
+
     __match_args__ = ('value',)
 
     @property
-    def value(self) -> _T: ...
+    def value(self) -> _T:
+        """
+        Returns the value wrapped by `Some`.
+        """
     @classmethod
     def __class_getitem__(cls, __item: Any) -> Type[Self]: ...
 
 @final
 class SchemaValidator:
-    def __new__(cls, schema: CoreSchema, config: CoreConfig | None = None) -> Self: ...
+    """
+    `SchemaValidator` is the Python wrapper for `pydantic-core`'s Rust validation logic, internally it owns one
+    `CombinedValidator` which may in turn own more `CombinedValidator`s which make up the full schema validator.
+    """
+
+    def __new__(cls, schema: CoreSchema, config: CoreConfig | None = None) -> Self:
+        """
+        Create a new SchemaValidator.
+
+        Arguments:
+            schema: The [`CoreSchema`][pydantic_core.core_schema.CoreSchema] to use for validation.
+            config: Optionally a [`CoreConfig`][pydantic_core.core_schema.CoreConfig] to configure validation.
+        """
```

Comment:
It looks like `__new__` doesn't show in the docs. @Kludex also reported an issue with `__new__` and `__init__` mismatching arguments in some cases (e.g. for exceptions, `__init__` comes from `BaseException`). I think this is probably a PyO3 bug, PyO3 sets `__new__` but not `__init__`.

Maybe for now it's more practical to set both `__new__` and `__init__` here, document `__init__`, and I'll have a think about how to resolve this in PyO3?